torch/_higher_order_ops/flex_attention.py

```diff
@@ -708,7 +708,8 @@ def flex_attention_autograd(
 
     with TransformGetItemToIndex():
         input_requires_grad = any(
-            t.requires_grad for t in (query, key, value, *score_mod_other_buffers)
+            isinstance(t, torch.Tensor) and t.requires_grad
```

Comment:
were you hitting this??

Reply:
Yes, the bias grads breaks score mods with captured dynamic shapes (like alibi).

Reply:
score_mod_other_buffers contains sympy expressions in that case?

Reply:
@drisspg yeah